overlord/snapstate/booted.go

```diff
@@ -53,11 +53,19 @@ func UpdateBootRevisions(st *state.State) error {
 		return nil
 	}
 
-	kernel, err := boot.GetCurrentBoot(snap.TypeKernel)
+	deviceCtx, err := DeviceCtx(st, nil, nil)
+	if err != nil {
+		if err == state.ErrNoState {
+			// too early anyway, no model
+			return nil
+		}
+	}
+
```

Comment:
here you're carrying on with a bad (nil?) deviceCtx, which will AFAIK crash when you try to use methods on it via the boot.Device interface. Suspect you meant to `return err`.
Also suspect you're missing a test :-)

Reply:
you were right, also I was a bit overthinking this, if we have a kernel we should have a model, so errors shouldn't not ever be ignored here